concourse/replicator.py

```diff
@@ -38,6 +39,12 @@
 logger = logging.getLogger(__name__)
 
 
+class RenderOrigin(enum.Enum):
```

Comment:
tbh, I think in this case, a `str` might actually a better choice (allows for more flexibility); the actual value is in fact only intended to be understood/used by humans, is it not? there is of course not hard reason against using an enum

Reply:
Handling enums is a bit less error prone than using strings directly, I think.
I'll keep the enum (with an added `UNKNOWN`